packages/widget/src/context/config.ts

```diff
@@ -0,0 +1,72 @@
+import { customElement, property } from 'lit/decorators.js';
+import { createContext, provide } from '@lit/context';
+import type { WalletConnectOptions } from '@web3-onboard/walletconnect/dist/types';
+import type { HTMLTemplateResult } from 'lit';
+import { html } from 'lit';
+import type { AppMetadata } from '@web3-onboard/common';
+import { BaseComponent } from '../components/common/base-component';
+import type { Theme } from '../interfaces';
+
+export interface ConfigContext {
+  theme?: Theme;
+  walletConnectOptions?: WalletConnectOptions;
+  appMetaData?: AppMetadata;
+}
+
+export const configContext = createContext<ConfigContext>(
+  Symbol('sygma-config-context')
+);
+
+declare global {
+  interface HTMLElementEventMap {
+    configUpdate: ConfigUpdateEvent;
+  }
+}
+
+export class ConfigUpdateEvent extends CustomEvent<Partial<ConfigContext>> {
+  constructor(update: Partial<ConfigContext>) {
+    super('configUpdate', { detail: update, composed: true, bubbles: true });
+  }
+}
+
+@customElement('sygma-config-context-provider')
+export class ConfigContextProvider extends BaseComponent {
+  @provide({ context: configContext })
+  private configContext: ConfigContext = {};
+
+  @property({ attribute: false, type: Object })
+  walletConnectOptions?: WalletConnectOptions;
+
+  @property({ attribute: false, type: Object })
+  appMetadata?: AppMetadata;
+
+  @property({ attribute: false, type: Object })
+  theme?: Theme;
```

Comment:
you should hook into property update and update context. That way if widget user updates theme it gets propagated (for example switching between light and dark theme).

Example of listening on property updates: https://github.com/sygmaprotocol/sygma-widget/blob/main/packages/widget/src/components/common/buttons/connect-wallet.ts#L42

Alternative is to do it in render but it seems like bad practice

Reply:
@mpetrunic implemented. I thought I would need to use force for this, since there is an update on an object, but here we are not dealing with deep property update